libs/ml/examples/cbow_simple/main.cpp

```diff
@@ -0,0 +1,333 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <math.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "math/approx_exp.hpp"
+#include "math/tensor.hpp"
+#include "polyfill.hpp"
+#include "unigram_table.hpp"
+#include "w2v_cbow_dataloader.hpp"
+#include "word_loader.hpp"
+
+using namespace std::chrono;
+using namespace fetch::ml;
+using FloatType = float;
+
+CBOWLoader<FloatType> data_loader(5, 25);
+WordLoader<FloatType> new_loader;
```

Comment:
please correct me if i'm wrong but it looks like this WordLoader is only instantiated here and never used; and then the whole WordLoader class is never used anywhere else?